### PR TITLE
[SMC-20] Streamline data upload and submission of sfkit workflows

### PR DESCRIPTION
**Pull request by dinvlad**
## Key Changes

1. **Terra Integration**
   - Adds support for Terra workspaces, including workspace selection and direct workspace bucket data upload.
   - Modifies instructions and UI to accommodate launching Terra workflows.

2. **Compute Environment Configuration**
   - Does away with per-study UI choice for "auto-configured" or "user-configured" studies, in favor of per-user choice for any given study. In other words, we let each user decide if they want sfkit service to automatically start user workflow (or raw GCP VM on non-Terra) when they hit "Begin Workflow" button, or to follow the instructions to download their pet SA key (or auth key on non-Terra) and run sfkit command to start the protocol.
   - Consequently, renames Configure Study section into Configure Compute Environment for clarity, and updates compute environment configuration process for both Terra and non-Terra environments.

3. **Authentication and Security**
   - Replaces Download Auth Key button with Download Service Account Key on Terra. This uses Sam's API to download user's pet SA key from Sam directly in the browser.
   - Adds `client_id` from `appConfig.json` to B2C auth scopes. Apparently, in some cases (like in local and non-Terra dev testing environments) without this scope B2C returns only `id_token`, but not `access_token`: https://learn.microsoft.com/en-us/answers/questions/135912/azure-ad-b2c-access-token-missing.
   - Fixes Terra hostname regex. Previously, it mistakenly compared `apiBaseUrl` against `sfkitui.dsde-(dev|staging|prod).broadinstitute.org` and `sfkit.terra.bio` hostname patterns (in https://github.com/hcholab/sfkit-react/pull/50), but those are UI hosts.

4. **Code Development and Linting**
   - Upgrades ESLint configuration to new flat config format.
   - Adds `vite-plugin-checker` for development-time linter checks.
   - Uses dev Rawls and Sam API URLs for local dev environment.

5. **Documentation**
    - Updates Instructions page for both Terra and non-Terra environments.
    - Disables Tutorials section on Terra in order to reduce possible confusion for Terra users, since it requires a bit more work than the already large scope of this PR.

## Impact

These changes significantly enhance the application's functionality by integrating Terra support, improving the user experience, and streamlining the study configuration and execution process. The PR also includes various code quality improvements and some dependency updates.

Since all changes are on the client-side, the suggested security impact is Low. We reuse existing sfkit backend APIs, as well as add calls to a few Terra and GCS APIs, which mirror similar calls made in the regular Terra UI, for data setup, upload and workflow launching. 

https://broadworkbench.atlassian.net/browse/SMC-20

### CODE REVIEW

File: src/components/studies/InstructionArea.tsx [hunk 1]
```diff
@@ -194,15 +188,40 @@ const InstructionArea: React.FC<Props> = ({
...
+              </>
+            )} will also need to download <code>
+            { onTerra ? "service_account_key.json" : "auth_key.txt" }
+            </code> { onTerra && "and run the following command " }
+            to authenticate the <i>sfkit</i> command-line interface:
+          </p>
+          { onTerra && (
+            <p className="p-2 rounded" style={{ backgroundColor: "#f0f0f0" }}>
+              <code>
+                export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account_key.json
```
Comment (sarahgibs) [inline src/components/studies/InstructionArea.tsx:208]: This needs to come with a warning that these need to be protected as a secret. As the key is long lived and probably shouldn't be used at all if you can help it.

File: src/pages/studies/Study.tsx [hunk 2]
```diff
@@ -91,31 +91,32 @@ const Study: React.FC = () => {
...
-    } catch (error) {
-      console.error("Failed to download auth key:", error);
+      document.body.removeChild(a);
+      URL.revokeObjectURL(objUrl);
+    } catch (err) {
+      console.error(`Failed to download ${fileName}:`, err);
     }
   };
 
+  const handleDownloadSAKey = () =>
```
Comment (sarahgibs) [inline src/pages/studies/Study.tsx:114]: Is this stored in local or session storage? And can instead this be handled by a token and refresh token? 
Session storage is more brittle, but safer for long lived secrets.

Comment (dinvlad) [reply src/pages/studies/Study.tsx:114]: Luckily, not store anywhere in the browser, and is just requested and downloaded on demand by the user

File: src/components/studies/InstructionArea.tsx [hunk 3]
```diff
@@ -194,15 +188,48 @@ const InstructionArea: React.FC<Props> = ({
...
+            <>
+              <p className="p-2 rounded" style={{ backgroundColor: "#f0f0f0" }}>
+                <code>
+                  export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service_account_key.json
+                </code>
+              </p>
+              <div className="alert alert-warning mt-2">
+                <strong>Warning:</strong> This key contains sensitive credentials.
+                Store it in a secure location on your computer, such as
+                the <code>~/.config/gcloud/</code> directory.
```
Comment (TomConner) [inline src/components/studies/InstructionArea.tsx:215]:
```suggestion
                Store it in an out-of-the-way location on your computer, such as
                the <code>~/.config/gcloud/</code> directory.
```

**Review by TomConner — APPROVED** [review]

**Review by zbedo — APPROVED** [review]

**Pull request merged**